File: app/src/main/java/com/stochanskyi/itunescharts/presentation/base/MVP/BasePresenter.java
package com.stochanskyi.itunescharts.presentation.base.MVP;

public abstract class BasePresenter<V extends IBaseView> implements IBasePresenter<V> {
    protected V view;

    @Override
    public void attachView(V view) {
        this.view=view;
    }

    @Override
    public void release() {
        this.view = null;
    }
}

File: app/src/main/java/com/stochanskyi/itunescharts/presentation/base/mvp/BasePresenter.kt
package com.stochanskyi.itunescharts.presentation.base.mvp

import androidx.annotation.CallSuper

abstract class BasePresenter<T : IBaseView>: IBasePresenter<T> {
    override var view: T? = null

    override fun onCreate() {}

    override fun onStart() {}

    override fun onPause() {}

    override fun onResume() {}

    @CallSuper
    override fun release() {
        view = null
    }
}